crate/pkcs11/provider/src/pkcs11_data_object.rs

```diff
@@ -0,0 +1,55 @@
+use std::ffi::CString;
+
+use cosmian_pkcs11_module::traits::DataObject;
+use sha3::Digest;
+use zeroize::{Zeroize, Zeroizing};
+
+use crate::{error::Pkcs11Error, kms_object::KmsObject};
+
+/// A PKCS11 data object is a DataObject that wraps data from a KMS object
+#[derive(Debug)]
+pub struct Pkcs11DataObject {
+    pub value: Zeroizing<Vec<u8>>,
+    pub label: String,
+}
+
+impl TryFrom<KmsObject> for Pkcs11DataObject {
+    type Error = Pkcs11Error;
+
+    fn try_from(kms_object: KmsObject) -> Result<Self, Self::Error> {
+        Ok(Pkcs11DataObject {
+            value: kms_object.object.key_block()?.key_bytes()?,
+            label: kms_object.other_tags.join(","),
+        })
+    }
+}
+
+impl Zeroize for Pkcs11DataObject {
+    fn zeroize(&mut self) {
+        self.value.zeroize();
+    }
+}
```

Comment:
Dead code ? It seems your object is not exported.

Reply:
The DataObject trait requires Zeroize